s3tests_boto3/functional/test_iam.py

```diff
@@ -2052,3 +2052,290 @@ def test_allow_deny_put_bucket_versioning_iam_policy_others():
     _delete_all_objects(s3_client_alt, bucket)
     response = s3_client_iam.delete_bucket(Bucket=bucket)
     eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Allow and Deny Get Object Version API using IAM policy for self')
+@attr(assertion='succeeds')
+@attr('user-policy')
+@attr('test_of_iam')
+def test_allow_deny_get_object_version_iam_policy_self():
+    client = get_iam_client()
+    s3_client_iam = get_iam_s3client()
+    bucket = get_new_bucket(client=s3_client_iam)
+    response = s3_client_iam.put_bucket_versioning(Bucket=bucket,
+                                                   VersioningConfiguration={"Status": "Enabled"})
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    s3_client_iam.put_object(Bucket=bucket, Key='iam1buk1obj1', Body='bar')
+    response = s3_client_iam.list_object_versions(Bucket=bucket, Prefix='iam1buk1obj1')
+    version_id = response['Versions'][0]['VersionId']
+    response = s3_client_iam.get_object(Bucket=bucket, Key='iam1buk1obj1', VersionId=version_id)
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    deny_get_object_versioning_policy = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Deny",
+             "Action": ["s3:GetObjectVersion"],
+             "Resource": f"arn:aws:s3:::{bucket}"
+         }
+         }
+    )
+    response = client.put_user_policy(PolicyDocument=deny_get_object_versioning_policy,
+                                      PolicyName='deny_policy_versioning',
```

Comment:
Policy name can not have '_' (underscore) in it. Ref: [AWS - IAM - PutUserPolicy](https://docs.aws.amazon.com/IAM/latest/APIReference/API_PutUserPolicy.html)

Reply:
![image](https://user-images.githubusercontent.com/66411891/188628437-2fbe874a-11be-4e1e-9727-5fec579a3e5e.png)

Reply:
My bad. Pattern just below has \w (which includes `_`)
> Pattern: [\w+=,.@-]+